### PR TITLE
feat(astro-integration-angular): move ng_astro to analog monorepo

### LINKED ISSUE #24: [LIB]: Migrate Angular Integration for Astro to this repo
I've developed a plugin for Astro that let's use Angular components inside of an Astro project. The repository currently lives here https://github.com/brandonroberts/ng-astro in the `integration-angular` folder.

This effort would:

- Move that integration to this repo
- Rename the package to @analogjs/astro-integration-angular
- Update the package to support the new `createApplication` API upcoming in Angular 14.2 (TBD)

### PR DESCRIPTION
**Pull request by mainawycliffe**
Work in Progress 

Closes #24

### CODE REVIEW

**Comment by brandonroberts** [comment]
You don't need to bring over the Astro Blog example app. Will probably create something else from scratch that looks a bit better

**Comment by brandonroberts** [comment]
And would prefer to convert the source files to TypeScript and compile them in the build step

**Comment by mainawycliffe** [comment]
Will remove them, for now I need something to test the integration with, will clean up afterwards, probably shouldn't have merge that in

**Comment by mainawycliffe** [comment]
In terms of converting to TS, I will do that, hoping to wrap this over the weekend.

**Comment by brandonroberts** [comment]
👍 As you were 😂

**Comment by mainawycliffe** [comment]
I think I have everything setup correctly, just need to figure out testing now

**Comment by brandonroberts** [comment]
Is this ready for review?

File: packages/astro-integration-angular/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  };
+}
+
+function getViteConfiguration() {
+  return {
+    optimizeDeps: {
+      include: ['@analogjs/astro-integration-angular/client.js'],
+      exclude: ['@analogjs/astro-integration-angular/server.js'],
+    },
+    plugins: [(viteAngular as any).default()],
```
Comment (brandonroberts) [inline packages/astro-integration-angular/src/index.ts:18]: This should just be `angular()` without any casting

File: packages/astro-integration-angular/package.json [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  "bugs": {
+    "url": "https://github.com/analogjs/analog/issues"
+  },
+  "homepage": "https://github.com/analogjs/analog#readme",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/analogjs/analog.git"
+  },
+  "dependencies": {
+    "@analogjs/vite-plugin-angular": "^0.2.0-alpha.0"
```
Comment (brandonroberts) [inline packages/astro-integration-angular/package.json:35]: Change to `latest` for now instead of the alpha version

**Review by brandonroberts — CHANGES_REQUESTED** [review]

File: packages/astro-integration-angular/package.json (hunk 2 above)
Comment (mainawycliffe) [reply packages/astro-integration-angular/package.json:35]: Yeah, this makes sense

File: packages/astro-integration-angular/src/index.ts (hunk 1 above)
Comment (mainawycliffe) [reply packages/astro-integration-angular/src/index.ts:18]: `angular()` works, but the plugin doesn't work with a project, as I am getting the following error `angular is not a function` when testing, I am guessing this has to do with the output of the vite plugin, which produces an export in the form of `exports.default` which seems incompatible with Astro integration which is ESM based. any ideas?

Comment (mainawycliffe) [reply packages/astro-integration-angular/src/index.ts:18]: I am guessing this is the reason you had the workaround to begin with

Comment (brandonroberts) [reply packages/astro-integration-angular/src/index.ts:18]: Yep, that was the reason. The vite plugin is currently shipped as commonjs. I think when I tried shipping ESM, it broke when used with Vite. Something more to investigate there, but I think we're good for now here.

Comment (mainawycliffe) [reply packages/astro-integration-angular/src/index.ts:18]: Let me add a comment to not this right now and then create an issue around this

**Review by brandonroberts — APPROVED** [review]
LGTM

**Pull request merged**